src/CMakeLists.txt

```diff
@@ -631,6 +634,16 @@ if(NOT WIN32 AND NOT APPLE)
     endif()
 endif()
 
+############################################################
+# MIOpen depends on libMLIRMIOpen for implicit gemm
+if(LIBMLIRMIOPEN)
+  MESSAGE(STATUS "LibMLIRMIOpen: " ${LIBMLIRMIOPEN})
+  set(LIBMLIRMIOPEN "${LIBMLIRMIOPEN}")
```

Comment:
What is this for? This variable is never set. Shouldn't you call `find_library` or `find_package`(preferred) to find and link mlir?

Reply:
Nevermind, I see the `find_library` call.

Reply:
This `set` still seems odd though.

Reply:
I'm fairly new on cmake let me know if there's a better way to do it.

I used to have the `find_library` called in this place, but not in root directory's `CMakeLists.txt` as I do now. The problem is that I also reuse the `LIBMLIRMIOpen` as the `#define` variable. What that means is that `find_library()` call has to go before the 

```configure_file("{SOURCE_DIR}/include/config.h.in" "{BINARY_DIR}/include/config.h"). ```

Unfortunately that `configure_file` line is on the top of the `src/CMakeLists.txt`. That's why I ended up having the `find_library()` call in the root directory's CMakeLists.txt, and set it at a later stage, in here.

Reply:
You should use two different variable names. One for `find_library` and another for `configure_file`.

Reply:
Will give that a shot, thanks!